core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -123,33 +130,35 @@ DeleteFile[] forEntry(ManifestEntry<DataFile> entry) {
   }
 
   DeleteFile[] forDataFile(long sequenceNumber, DataFile file) {
+    if (isEmpty) {
```

Comment:
No need to derive the partition and do anything with the streams if the index is empty.